src/daemon/globals.c

```diff
@@ -0,0 +1,13 @@
+#include "globals.h"
+#include "plugin.h"
```

Comment:
I think you need `"plugin.h"` for `DATA_MAX_NAME_LEN`, which is one of those defines that might actually make sense to move to here (and out of `"plugin.h"`).